pkg/operator/client/delete.go

```diff
@@ -79,6 +79,13 @@ func (c *Client) diffAndDeleteManifests(opts DiffAndDeleteOptions) error {
 			}
 		}
 
+		// if this is a keep resource, don't delete it
+		// e.g. for migration to Helm v1beta2
+		if keep, ok := o.Metadata.Annotations["kots.io/keep"]; ok && keep == "true" {
+			logger.Infof("skipping deletion of resource %s/%s", o.Kind, o.Metadata.Name)
```

Comment:
would be nice to mention the reason in the log message as well
```suggestion
			logger.Infof("skipping deletion of resource %s/%s because kots.io/keep annotation is set", o.Kind, o.Metadata.Name)
```